### PR TITLE
revises network portion of 2.5 porting guide

### LINKED ISSUE #36777: 2.5 Porting Guide to mention network_cli
##### ISSUE TYPE
 - Documentation Report

##### COMPONENT NAME
docs

##### ANSIBLE VERSION
```
2.5.0
```

##### SUMMARY
http://docs.ansible.com/ansible/devel/porting_guides/porting_guide_2.5.html#networking 
* Some details about moving from `connection: local` to `network_cli`
* Link to new network docs

### PR DESCRIPTION
**Pull request by acozine**
##### SUMMARY
Adds more detail, including the persistent connection options to the Network section of the 2.5 Porting Guide.

Fixes #36777

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
- Network Documentation

##### ANSIBLE VERSION
2.5 and higher

### CODE REVIEW

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 1]
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
...
+
+.. code-block:: ini
+
+   [ios:vars]
+   ansible_become=yes
+   ansible_become_method=enable
+   ansible_become_pass=cisco
+   ansible_network_os=ios
+   ansible_password=cisco
+   ansible_user=cisco
```
Comment (gundalow) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: Worth linking to network docs here again?

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 2]
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
...
 
-Will result in:
+Adding persistent connection types ``network_cli`` and ``netconf``
+------------------------------------------------------------------
+
+Ansible 2.5 introduces two persistent connection types, ``network_cli`` and ``netconf``. We recommend you use these whenever possible.
+Unless you need a ``local`` connection, update your playbooks to use ``network_cli`` or ``netconf`` and to
```
Comment (gundalow) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: Maybe clarify that `local` is needed when using eAPI/NX-API?

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 3]
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
...
 
-Will result in:
+Adding persistent connection types ``network_cli`` and ``netconf``
+------------------------------------------------------------------
+
+Ansible 2.5 introduces two persistent connection types, ``network_cli`` and ``netconf``. We recommend you use these whenever possible.
+Unless you need a ``local`` connection, update your playbooks to use ``network_cli`` or ``netconf`` and to 
+specify your connection variables with standard Ansible connection variables:
+
+OLD In Ansible 2.4
```
Comment (gundalow) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: `**OLD** In Ansible 2.4`

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 4]
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
...
+   2.9. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
+   [DEPRECATION WARNING]: Param 'password' is deprecated. See the module docs for more information. This feature will be removed in version
+   2.9. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
+   [DEPRECATION WARNING]: Param 'host' is deprecated. See the module docs for more information. This feature will be removed in version 2.9.
+   Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
+
+We recommend setting connection properties in inventory by group. As you update your playbooks and inventory files, you can easily make the change to ``become`` for privilege escalation (on platforms that support it). For more information, see the :ref:`using become with network modules<become-network>` guide and the :ref:`platform documentation<platform_options>`. To update the ``ios_command`` task above, move the connection properties to inventory like this:
+
+NEW In Ansible 2.5
```
Comment (gundalow) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: `**NEW** In Ansible 2.5`

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 5]
```diff
@@ -199,12 +199,16 @@ Porting custom scripts
...
-==========
+Network
+=======
+
+Top-level connection arguments will be removed in 2.9
+-----------------------------------------------------
 
+Top-level connection arguments like ``username``, ``host``, and ``password`` are deprecated and will be removed in version 2.9.
+
+OLD In Ansible 2.4
```
Comment (gundalow) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: `**OLD** In Ansible 2.4`

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 6]
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
...
+          transport: cli
+
+   tasks:
+   - nxos_config:
+       src: config.j2
+       provider: "{{ cli }}"
+       username: admin
+       password: admin
 
+NEW In Ansible 2.5
```
Comment (gundalow) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst]: `**NEW** In Ansible 2.5`

**Comment by acozine** [comment]
@gundalow I addressed your comments in b3ff9bb8f014a6f29f5e0d8f16f3405a45d361cf - let me know if there are more changes you'd like to see, more links, more emphasis, etc.

**Review by ganeshrn — APPROVED** [review]

**Review by trishnaguha — APPROVED** [review]

File: docs/docsite/rst/porting_guides/porting_guide_2.5.rst [hunk 7]
```diff
@@ -199,12 +199,16 @@ Porting custom scripts
 
 No notable changes.
 
-Networking
-==========
+Network
```
Comment (gundalow) [inline docs/docsite/rst/porting_guides/porting_guide_2.5.rst:202]: I think somewhere prominent in this section we should link to the network guide.

**Review by gundalow — APPROVED** [review]

**Comment by acozine** [comment]
Chatted with Scott today - he says "if it looks good to you I'm fine with it." I'm going to take that as a positive review and merge.

**Comment by gundalow** [comment]
In `stable-2.5` so marking as done

**Pull request merged**